File: app/src/main/java/com/example/domain/repository/PfmsRepository.kt
package com.example.domain.repository

import com.example.core.database.entities.PersonDebtAccountEntity
import com.example.domain.model.*
import kotlinx.coroutines.flow.Flow

interface PfmsRepository {
    fun getAccounts(): Flow<List<Account>>
    fun getTransactions(): Flow<List<Transaction>>
    fun getTransactionsForAccount(accountId: String): Flow<List<Transaction>>
    fun getAssets(): Flow<List<Asset>>
    fun getDebts(): Flow<List<Debt>>
    fun getBudgets(): Flow<List<Budget>>
    fun getGoals(): Flow<List<Goal>>
    fun getBills(): Flow<List<Bill>>

    suspend fun addAccount(account: Account)
    suspend fun updateAccount(account: Account)
    suspend fun archiveAccount(accountId: String)
    suspend fun addTransaction(transaction: Transaction)
    suspend fun updateTransaction(transaction: Transaction)
    suspend fun deleteTransaction(id: String)
    suspend fun addAsset(asset: Asset)
    suspend fun updateAsset(asset: Asset)
    suspend fun deleteAsset(id: String)
    fun getAssetLogs(assetId: String): Flow<List<AssetLog>>
    suspend fun addAssetLog(log: AssetLog)

    fun getPersons(): Flow<List<Person>>
    suspend fun addPerson(person: Person)
    suspend fun updatePerson(person: Person)

    fun getDebtAccounts(): Flow<List<PersonDebtAccountEntity>>
    suspend fun addDebtAccount(account: PersonDebtAccountEntity)
    suspend fun updateDebtAccount(account: PersonDebtAccountEntity)

    suspend fun addDebt(debt: Debt)
    suspend fun recordDebtPayment(debtId: String, paymentAmount: Double, accountId: String)
    suspend fun deleteDebt(id: String)

    suspend fun addBudget(budget: Budget)
    suspend fun deleteBudget(id: String)

    suspend fun addGoal(goal: Goal)
    suspend fun contributeToGoal(goalId: String, amount: Double, accountId: String)
    suspend fun deleteGoal(id: String)

    suspend fun addBill(bill: Bill)
    suspend fun payBill(billId: String, accountId: String)
    suspend fun deleteBill(id: String)

    suspend fun seedInitialSampleDataIfEmpty()
    suspend fun isAccountInUse(accountId: String): Boolean
    suspend fun deleteAccount(accountId: String)
}

File: lib/domain/repository/pfms_repository.dart
import '../models/domain_models.dart';
import '../models/debt_models.dart';
import '../models/person.dart';

abstract class PfmsRepository {
  Stream<List<Account>> getAccounts();
  Stream<List<Transaction>> getTransactions();
  Stream<List<Transaction>> getTransactionsForAccount(String accountId);
  Stream<List<Asset>> getAssets();
  Stream<List<Debt>> getDebts();
  Stream<List<Budget>> getBudgets();
  Stream<List<Goal>> getGoals();
  Stream<List<Bill>> getBills();
  Stream<List<Person>> getPersons();
  Stream<List<AssetLog>> getAssetLogs(String assetId);

  Future<void> addAccount(Account account);
  Future<void> updateAccount(Account account);
  Future<void> archiveAccount(String accountId);
  Future<void> deleteAccount(String accountId);
  Future<bool> isAccountInUse(String accountId);

  Future<void> addTransaction(Transaction transaction);
  Future<void> updateTransaction(Transaction transaction);
  Future<void> deleteTransaction(String id);

  Future<void> addAsset(Asset asset);
  Future<void> updateAsset(Asset asset);
  Future<void> deleteAsset(String id);
  Future<void> addAssetLog(AssetLog log);

  Future<void> addPerson(Person person);
  Future<void> updatePerson(Person person);

  Future<void> addDebt(Debt debt);
  Future<void> recordDebtPayment(String debtId, double paymentAmount, String accountId);
  Future<void> deleteDebt(String id);

  Future<void> addBudget(Budget budget);
  Future<void> deleteBudget(String id);

  Future<void> addGoal(Goal goal);
  Future<void> contributeToGoal(String goalId, double amount, String accountId);
  Future<void> deleteGoal(String id);

  Future<void> addBill(Bill bill);
  Future<void> payBill(String billId, String accountId);
  Future<void> deleteBill(String id);

  Future<void> seedInitialSampleDataIfEmpty();
}
